protocol/kafkaapi/api_versions_response.go

```diff
@@ -0,0 +1,147 @@
+package kafkaapi
+
+import (
+	"fmt"
+
+	"github.com/codecrafters-io/kafka-tester/protocol/decoder"
+	"github.com/codecrafters-io/kafka-tester/protocol/errors"
+	"github.com/codecrafters-io/kafka-tester/protocol/kafkaapi/headers"
+	"github.com/codecrafters-io/tester-utils/logger"
+)
+
+type ApiVersionsResponse struct {
+	Header headers.ResponseHeader
+	Body   ApiVersionsResponseBody
+}
+
+func (r *ApiVersionsResponse) Decode(response []byte, logger *logger.Logger) (err error) {
+	decoder := decoder.NewDecoder(response, logger)
+
+	decoder.BeginSubSection("ApiVersions Response")
+
+	defer func() {
+		decoder.EndCurrentSubSection()
```

Comment:
Let's place this above as `defer decoder.EndCurrentSubSection`, `defer` calls are easier to read when they're placed close to the action they're reversing